cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/ApiCredentials.java

```diff
@@ -0,0 +1,328 @@
+     */
+    /* test */ static Map<String, UserEntry> generateUserManagedApiCredentials(Secret secret, String secretKey) {
+        Map<String, UserEntry> entries = new HashMap<>();
+        if (secret != null) {
+            if (secretKey != null && secret.getData().containsKey(secretKey)) {
+                String credentialsAsString = Util.decodeFromBase64(secret.getData().get(secretKey));
+                entries.putAll(parseEntriesFromString(credentialsAsString));
+            }
+        }
+        for (UserEntry entry : entries.values()) {
+            if (FORBIDDEN_USERNAMES.contains(entry.username())) {
+                throw new InvalidConfigurationException("The following usernames for Cruise Control API are forbidden: " + FORBIDDEN_USERNAMES
+                        + " User provided Cruise Control API credentials contain illegal username: " + entry.username());
+            } else if (entry.role() == Role.ADMIN) {
+                throw new InvalidConfigurationException("The following roles for Cruise Control API are forbidden: " + Role.ADMIN
+                        + " User provided Cruise Control API credentials contain contains illegal role: " +  entry.role());
+            }
+        }
+        return entries;
+    }
+
+    /**
+     * Parses auth data from existing Cluster Operator managed API user secret into map of API user entries.
+     *
+     * @param passwordGenerator The password generator for API users
+     * @param secret API user secret
+     *
+     * @return Map of API user entries containing Strimzi-managed API user credentials
+     */
+    /* test */ static Map<String, UserEntry> generateCoManagedApiCredentials(PasswordGenerator passwordGenerator, Secret secret) {
+        Map<String, UserEntry> entries = new HashMap<>();
+
+        if (secret != null) {
+            if (secret.getData().containsKey(AUTH_FILE_KEY)) {
+                String credentialsAsString = Util.decodeFromBase64(secret.getData().get(AUTH_FILE_KEY));
+                entries.putAll(parseEntriesFromString(credentialsAsString));
+            }
+        }
+
+        if (!entries.containsKey(REBALANCE_OPERATOR_USERNAME)) {
+            entries.put(REBALANCE_OPERATOR_USERNAME, new UserEntry(REBALANCE_OPERATOR_USERNAME, passwordGenerator.generate(), ApiCredentials.Role.ADMIN));
+        }
+
+        if (!entries.containsKey(HEALTHCHECK_USERNAME)) {
+            entries.put(HEALTHCHECK_USERNAME, new UserEntry(HEALTHCHECK_USERNAME, passwordGenerator.generate(), ApiCredentials.Role.USER));
+        }
+
+        return entries;
+    }
+
+    /**
+     * Creates map with API usernames, passwords, and credentials file for Strimzi-managed API users secret.
+     *
+     * @param passwordGenerator the password generator used for creating new credentials.
+     * @param oldCruiseControlApiSecret the existing Cruise Control API secret, containing previously stored credentials.
```

Comment:
is this really the "old" or just the current Cruise Control API secret. I mean, reading "old" I would expect you are going to generate something "new", while you are just this Secret just to generate the corresponding current Api credentials, or?

I see you have `generateApiSecret` which gets the "old" and generate the "new" Secret, but I think that is where "old" makes sense. This `generateMapWithApiCredentials` is just generating a Map from a Secret you pass it. It doesn't matter it's old or new. I hope you see what I mean :-)
So in this case for me the parameter name would just be `cruiseControlApiSecret` here.